### ISSUE TITLE
only when override StateProvider, MyApp wont be rebuild, but update the state.

### ISSUE THREAD

**Issue by airy-swift** [rrousselGit/river_pod#771]
flutter_river_pod

**Describe the bug**
no problem circumstance: didnt override StateProvider.
the problem shown: when override StateProvider.

**To Reproduce**

```dart
void main() {
  testWidgets('update the UI when incrementing the state', (tester) async {
    await tester.pumpWidget(ProviderScope(
      /// if remove below 3lines, this test will pass.
      overrides: [
        counterProvider.overrideWithValue(StateController<int>(0)),
      ],
      child: MyApp(),
    ));

    // The default value is `0`, as declared in our provider
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Increment the state and re-render
    await tester.tap(find.byType(ElevatedButton));
    await tester.pump();

    // The state have properly incremented
    expect(find.text('1'), findsOneWidget); /// <- '1' not found error
    expect(find.text('0'), findsNothing);
  });

  testWidgets('the counter state is not shared between tests', (tester) async {
    await tester.pumpWidget(MyAppWrapper());

    // The state is `0` once again, with no tearDown/setUp needed
    expect(find.text('10'), findsOneWidget);
    expect(find.text('11'), findsNothing);
  });
}

```

**Expected behavior**
the test will pass.

thanks for your read this issue.
if my confirmation your specification was less , verrrry sorrrry ~~٩( 'ω' )و

**Comment by rrousselGit** [comment]
Hello!

Could you tell me which version you're using? I believe this is a problem that was fixed in the dev release, but it may not be fixed in the version 0.14.x

**Comment by airy-swift** [comment]
@rrousselGit 

thanks early reply!

oh really sorry, i'm using ^0.14.0+2!
i'll update! thanks!
